From 73d4cbe355bc8d0394f61694ac4e16c0ceaed288 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Fri, 3 May 2024 17:14:36 +0200
Subject: [PATCH 0820/1695] dts: stm32: add a default backup register
 configuration on stm32mp15 boards

The "st,backup-zones" property is mandatory, add a default configuration
at SoC level. It can be superseded by a board configuration.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 0e0df2dfe02d2a9b4927b89498b8986340f64252 dts: stm32: add a default backup register configuration on stm32mp15 boards]
Change-Id: Ibb35a3948f146fbb6eeef105e8f68bede6d034ab
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/378284
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/378786
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/dts/stm32mp151.dtsi      | 1 +
 core/arch/arm/dts/stm32mp157a-ed1.dts  | 4 ----
 core/arch/arm/dts/stm32mp157c-ed1.dts  | 4 ----
 core/arch/arm/dts/stm32mp157d-ed1.dts  | 4 ----
 core/arch/arm/dts/stm32mp157f-ed1.dts  | 4 ----
 core/arch/arm/dts/stm32mp15xx-dkx.dtsi | 2 --
 6 files changed, 1 insertion(+), 18 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp151.dtsi b/core/arch/arm/dts/stm32mp151.dtsi
index 0d8683bd5..a06caf2cb 100644
--- a/core/arch/arm/dts/stm32mp151.dtsi
+++ b/core/arch/arm/dts/stm32mp151.dtsi
@@ -1644,6 +1644,7 @@
 			interrupts = <GIC_SPI 197 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&rcc RTCAPB>;
 			ranges;
+			st,backup-zones = <10 5 17>;
 
 			nvram: nvram@5c00a100 {
 				compatible = "st,stm32mp15-tamp-nvram";
diff --git a/core/arch/arm/dts/stm32mp157a-ed1.dts b/core/arch/arm/dts/stm32mp157a-ed1.dts
index a54b5dad4..bf89943aa 100644
--- a/core/arch/arm/dts/stm32mp157a-ed1.dts
+++ b/core/arch/arm/dts/stm32mp157a-ed1.dts
@@ -375,10 +375,6 @@
 	status = "okay";
 };
 
-&tamp {
-	st,backup-zones = <10 5 17>;
-};
-
 &timers15 {
 	status = "okay";
 
diff --git a/core/arch/arm/dts/stm32mp157c-ed1.dts b/core/arch/arm/dts/stm32mp157c-ed1.dts
index 1443b25d1..7d8178075 100644
--- a/core/arch/arm/dts/stm32mp157c-ed1.dts
+++ b/core/arch/arm/dts/stm32mp157c-ed1.dts
@@ -570,10 +570,6 @@
 	status = "disabled";
 };
 
-&tamp {
-	st,backup-zones = <10 5 17>;
-};
-
 &timers6 {
 	status = "disabled";
 	/* spare dmas for other usage */
diff --git a/core/arch/arm/dts/stm32mp157d-ed1.dts b/core/arch/arm/dts/stm32mp157d-ed1.dts
index 8a2ea5d5b..7376f9ad7 100644
--- a/core/arch/arm/dts/stm32mp157d-ed1.dts
+++ b/core/arch/arm/dts/stm32mp157d-ed1.dts
@@ -374,10 +374,6 @@
 	status = "okay";
 };
 
-&tamp {
-	st,backup-zones = <10 5 17>;
-};
-
 &timers15 {
 	status = "okay";
 
diff --git a/core/arch/arm/dts/stm32mp157f-ed1.dts b/core/arch/arm/dts/stm32mp157f-ed1.dts
index 4f25bd008..bcf3daf47 100644
--- a/core/arch/arm/dts/stm32mp157f-ed1.dts
+++ b/core/arch/arm/dts/stm32mp157f-ed1.dts
@@ -375,10 +375,6 @@
 	status = "okay";
 };
 
-&tamp {
-	st,backup-zones = <10 5 17>;
-};
-
 &timers15 {
 	status = "okay";
 
diff --git a/core/arch/arm/dts/stm32mp15xx-dkx.dtsi b/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
index d46a4574c..5d4144030 100644
--- a/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
+++ b/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
@@ -742,8 +742,6 @@
 };
 
 &tamp {
-	status = "okay";
-	st,backup-zones = <10 5 17>;
 	st,tamp-passive-nb-sample = <4>;
 	st,tamp-passive-sample-clk-div = <16384>;
 };
-- 
2.47.2

